packages/fuels-core/src/types/wrappers/transaction.rs

```diff
@@ -75,7 +76,13 @@ pub trait Transaction: Into<FuelTransaction> + Send {
         parameters: &ConsensusParameters,
     ) -> Result<(), Error>;
 
-    fn id(&self, params: &ConsensusParameters) -> Bytes32;
+    fn precompute(&mut self, chain_id: u64) -> Result<(), Error>;
+
+    fn is_computed(&self) -> bool;
+
+    fn estimate_predicates(&mut self, parameters: &ConsensusParameters) -> Result<(), Error>;
+
+    fn id(&self, chain_id: u64) -> Bytes32;
```

Comment:
Same here

```suggestion
    fn id(&self, chain_id: ChainId) -> Bytes32;
```

Reply:
We don't want the user to have to worry about the `ChainId` type, same as with `BlockHeight`k